### PR TITLE
Docker support

### PR DESCRIPTION
**Pull request by tpsjr7**
Able to run the app from docker. Should help people who know how to use it get a working environment quickly. I've tested on Windows 10. Avoids a lot of the trouble of making sure certain libraries and CUDA are installed, etc.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,8 @@
+__pycache__/
```
Comment (tpsjr7) [inline .gitignore:1]: ignoring lots of generated artifacts from when the PyPatchMatch is compiled

File: docker/opencv.pc [hunk 2]
```diff
@@ -0,0 +1,11 @@
+prefix=/usr
```
Comment (tpsjr7) [inline docker/opencv.pc:1]: this file is to allow PyPatchMatch to find opencv

File: environment.yml [hunk 3]
```diff
@@ -235,4 +235,4 @@ dependencies:
   - zlib=1.2.12=h5eee18b_3
   - zstd=1.5.2=ha4553b6_0
   - pip:
-    - opencv-python==4.6.0.66
\ No newline at end of file
+    - opencv-python-headless==4.6.0.66
```
Comment (tpsjr7) [inline environment.yml:238]: the headless version is lighter doesn't need opencv's ui graphics, so allows it to run in docker. Appears to be able to run everything fine.

**Comment by lkwq007** [comment]
Thank you for the docker support!

**Comment by hunterunger** [comment]
Any idea why I keep getting the error `exec /home/user/app/docker/entrypoint.sh: no such file or directory` after running `docker-compose up`?

**Comment by tpsjr7** [comment]
> Any idea why I keep getting the error `exec /home/user/app/docker/entrypoint.sh: no such file or directory` after running `docker-compose up`?

maybe certain versions of docker don't like how I did the overlapping volumes, I'll try a different volume strategy. Mind sharing 
docker-compose -v and docker -v?

**Comment by hunterunger** [comment]
(Running on Windows 10)
```
$ docker-compose -v; docker -v
Docker Compose version v2.10.2
Docker version 20.10.17, build 100c701
```

**Comment by tpsjr7** [comment]
> (Running on Windows 10)
> 
> ```
> $ docker-compose -v; docker -v
> Docker Compose version v2.10.2
> Docker version 20.10.17, build 100c701
> ```
> 
> More details on running command:
> 
> ```
> $ docker-compose up
> [+] Running 4/4
>  - Network stablediffusion-infinity_default          Created                                                       0.7s
>  - Volume "stablediffusion-infinity_user_home"       Created                                                       0.0s
>  - Volume "stablediffusion-infinity_cond_env"        Created                                                       0.0s
>  - Container stablediffusion-infinity-sd-infinity-1  Created                                                       0.1s
> Attaching to stablediffusion-infinity-sd-infinity-1
> stablediffusion-infinity-sd-infinity-1  | exec /home/user/app/docker/entrypoint.sh: no such file or directory
> stablediffusion-infinity-sd-infinity-1 exited with code 1
> ```

A new PR https://github.com/lkwq007/stablediffusion-infinity/pull/49 was merged, can you update and try running it again?

**Pull request merged**